spec/tapioca/runtime/generic_type_registry_spec.rb

```diff
@@ -0,0 +1,103 @@
+# typed: true
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Tapioca
+  module Runtime
+    class GenericTypeRegistrySpec < Minitest::Spec
+      describe Tapioca::Runtime::GenericTypeRegistry do
+        describe ".generic_type_instance?" do
+          it "returns false for instances of non-generic classes" do
+            refute(GenericTypeRegistry.generic_type_instance?(Object.new))
+          end
+
+          it "returns true for instances of generic classes" do
+            assert(GenericTypeRegistry.generic_type_instance?(SampleGenericClass[Object].new))
+          end
+        end
+
+        describe ".lookup_type_variables" do
+          it "returns nil for non-generic types" do
+            assert_nil(GenericTypeRegistry.lookup_type_variables(Object))
+          end
+
+          it "returns the type variables for generic types" do
+            assert_equal([SampleGenericClass::Element], GenericTypeRegistry.lookup_type_variables(SampleGenericClass))
```

Comment:
Could we nest the test class in the test itself with a `sample_class = Class.new do ...`?

Reply:
https://github.com/Shopify/tapioca/pull/1635#discussion_r1313683273